### ISSUE TITLE
[Request] - SEGA Dreamcast .OSB, .MLT & .BGM Audio File Support

### ISSUE THREAD

**Issue by The-DC-OSB-MLT-Files** [vgmstream/vgmstream#1177]
I opened up a .ISO file of the game Illbleed and I found some obscure audio related file formats that have yet to be extracted or accessed in any working form successfully. 

The audio files are listed as .OSB (One Shot Bank) and .MLT. Both file types apparently contain archives of .ADX/.ADF/.ADPCM SEGA Dreamcast audio files. .BGM files are apparently supposed to contain the game's music data, but the former two file types seem to still remain inaccessible to this day, despite my best efforts to find a working solution.

I think adding support to extract/access SEGA's Dreamcast .OSB, .MLT & .BGM audio files would make for a great future update.

The files have been placed inside the attached .zip file for reference.

EDIT:

The .OSB and .MLT files can actually be played (although pretty poorly) through Audacity by selecting the option to input raw data and then using the following settings seems to get a barely working, broken result that can be heard.

Encoding: VOX ADPCM
Byte order: Doesn't seem to matter?
Channels: 1 Channel (Mono)
Sample rate: 22050Hz (Pitch sounds close to the original sounds in-game but not perfect - .MLT file could be different)

Aside from this, I haven't found any other way to use or extract these sound files cleanly and in working form.

[SEGA Dreamcast OSB MLT BGM Sample Files (From Illbleed).zip](https://github.com/vgmstream/vgmstream/files/9185635/SEGA.Dreamcast.OSB.MLT.BGM.Sample.Files.From.Illbleed.zip)
.

**Comment by bnnm** [comment]
Those files are mainly for sequenced audio, which vgmstream doesn't handle, so can't be really added.

You can play the .osb samples if you unzip this file in the same folder (may need to set "enable unknown exts" in foobar/winamp) and open the .osb: [txth.zip](https://github.com/vgmstream/vgmstream/files/9229349/txth.zip). Not sure how pitch/sample rate is calculated though.

.bgm is sequenced/midi data, while .mlt is a mix of .osb samples and other info. If you extract the .osb inside they should be playable with the above .txth.

**Comment by The-DC-OSB-MLT-Files** [comment]
I recently came across a toolkit that made it possible for me to export both the .OSB and .MLT files into .ADP files and then convert them into .WAV files. The tutorial that was provided made it ridiculously easy, which was awesome since I genuinely didn't expect anything to work in the end.

The pitch and sample rates do vary with each batch of sounds (some sound fine at 22050Hz whereas some require around a rate of 18000Hz to sound "correct") but they are all extracted and in full working order. No broken audio or anything that I can tell, aside from the mentioned sample rate issue.

Thanks for providing that file for me. If I can get it to work, I'll see if the results are different to what I did instead. I may as well go ahead and close this request now.

**Comment by riggles1** [comment]
> I recently came across a toolkit that made it possible for me to export both the .OSB and .MLT files into .ADP files and then convert them into .WAV files. The tutorial that was provided made it ridiculously easy, which was awesome since I genuinely didn't expect anything to work in the end.
> 

What toolkit was this? I used kingshriek's dsf tools to get OSB files out of the MLT container. But I can't get further than that, no idea how to turn the OSB into ADP/WAV. I can hear the soundeffects in Audacity.

The reason I'm looking into this is that I'm trying to make a Radirgy audio fix, that game is heavily distorted (unplayably distorted at stage 5 in particular), I just fixed the music using CD OST rips as a source (adjusting EQ and volume for no peaking). The resulting adx files turned out great.

However, the sound effects is in an MLT instead, and they're overly loud and distorted. I ripped the sound effects from the Gamecube version which has these intact and I want to insert those into the DC game (and possibly Naomi version too), but I'm stuck.

So any pointers are appreciated!
